tests/test_queue.py

```diff
@@ -581,3 +582,190 @@ def test_workflow(self, var1: str) -> str:
+
+    @DBOS.workflow()
+    def test_workflow() -> list[int]:
+        assert DBOS.workflow_id == wfid
+        handles = []
+        for i in range(queued_steps):
+            h = queue.enqueue(test_step, i)
+            handles.append(h)
+        return [h.get_result() for h in handles]
+
+    @DBOS.step()
+    def test_step(i: int) -> int:
+        nonlocal step_counter
+        step_counter += 1
+        step_events[i].set()
+        event.wait()
+        return i
+
+    # Start the workflow. Wait for all five steps to start. Verify that they started.
+    with SetWorkflowID(wfid):
+        original_handle = DBOS.start_workflow(test_workflow)
+    for e in step_events:
+        e.wait()
+    assert step_counter == 5
+
+    # Recover the workflow, then resume it.
+    recovery_handles = DBOS.recover_pending_workflows()
+    event.set()
+    # There should be one handle for the workflow and another for each queued step.
+    assert len(recovery_handles) == queued_steps + 1
+    # Verify that both the recovered and original workflows complete correctly.
+    for h in recovery_handles:
+        if h.get_workflow_id() == wfid:
+            assert h.get_result() == [0, 1, 2, 3, 4]
+    assert original_handle.get_result() == [0, 1, 2, 3, 4]
+    # Each step should start twice, once originally and once in recovery.
+    assert step_counter == 10
+
+    # Rerun the workflow. Because each step is complete, none should start again.
+    with SetWorkflowID(wfid):
+        assert test_workflow() == [0, 1, 2, 3, 4]
+    assert step_counter == 10
+
+    # Verify all queue entries eventually get cleaned up.
+    assert queue_entries_are_cleaned_up(dbos)
+
+
+def test_cancelling_queued_workflows(dbos: DBOS) -> None:
+    start_event = threading.Event()
+    blocking_event = threading.Event()
+
+    @DBOS.workflow()
+    def stuck_workflow() -> None:
+        start_event.set()
+        blocking_event.wait()
+
+    @DBOS.workflow()
+    def regular_workflow() -> None:
+        return
+
+    # Enqueue both the blocked workflow and a regular workflow on a queue with concurrency 1
+    queue = Queue("test_queue", concurrency=1)
+    wfid = str(uuid.uuid4())
+    with SetWorkflowID(wfid):
+        blocked_handle = queue.enqueue(stuck_workflow)
+    regular_handle = queue.enqueue(regular_workflow)
+
+    # Verify that the blocked workflow starts and is PENDING while the regular workflow remains ENQUEUED.
+    start_event.wait()
+    assert blocked_handle.get_status().status == WorkflowStatusString.PENDING.value
+    assert regular_handle.get_status().status == WorkflowStatusString.ENQUEUED.value
+
+    # Cancel the blocked workflow. Verify this lets the regular workflow run.
+    dbos.cancel_workflow(wfid)
+    assert blocked_handle.get_status().status == WorkflowStatusString.CANCELLED.value
+    assert regular_handle.get_result() == None
+
+    # Complete the blocked workflow
+    blocking_event.set()
+    assert blocked_handle.get_result() == None
```

Comment:
The status should also be `SUCCESS` here, right?

Reply:
It should complete to SUCCESS (and that's verified in other tests)